packages/legend-query-builder/src/stores/QueryBuilderState.ts

```diff
@@ -278,6 +281,13 @@ export abstract class QueryBuilderState {
     );
   }
 
+  async initializeQueryWithChangeDetection(
+    rawLambda: RawLambda,
+  ): Promise<void> {
+    this.initializeWithQuery(rawLambda);
+    await this.changeDetectionState.start();
```

Comment:
probably don't need to be async here, also, let's combine in a different way. I want everything to use `rebuildWithQueryBuilder` and have `initializeWithQuery` doing change detection as well